src/core/contracts/identity.test.ts

```diff
@@ -331,74 +339,358 @@ describe("identity", () => {
+
+      const delegateAddress = "0x70997970C51812dc3A010C7d01b50e0d17dc79C8";
+      const delegateLogData: DelegateLogData[] = [
+        {
+          name: "DSNPAddDelegate",
+          identityAddress: "0x71C95911E9a5D330f4D621842EC243EE1343292e",
+          delegate: "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+          blockNumber: 330,
+        },
+        {
+          name: "DSNPRemoveDelegate",
+          identityAddress: "0x71C95911E9a5D330f4D621842EC243EE1343292e",
+          delegate: "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+          blockNumber: 331,
+          endBlock: 400,
+        },
+      ];
+
+      it("returns delegate log data", async () => {
+        const result = await resolveDelegatesFor(delegateAddress, delegateLogData, { provider: provider });
+        expect(result).toEqual([delegateLogData[1]]);
+      });
+    });
+
+    describe("when a delegate is removed", () => {
+      const provider = ({
+        getBlockNumber: jest.fn().mockResolvedValue(333),
+      } as Partial<ethers.providers.Provider>) as ethers.providers.Provider;
+
+      it("returns zero delegate log data", async () => {
+        const delegateAddress = "0x70997970C51812dc3A010C7d01b50e0d17dc79C8";
+        const delegateLogData: DelegateLogData[] = [
+          {
+            name: "DSNPAddDelegate",
+            identityAddress: "0x8464135c8F25Da09e49BC8782676a84730C318bC",
+            delegate: "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+            blockNumber: 330,
+          },
+          {
+            name: "DSNPRemoveDelegate",
+            identityAddress: "0x8464135c8F25Da09e49BC8782676a84730C318bC",
+            delegate: "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+            blockNumber: 331,
+            endBlock: 0,
+          },
+        ];
+
+        expect(await resolveDelegatesFor(delegateAddress, delegateLogData, { provider: provider })).toEqual([]);
+      });
+    });
 
-        await upsertDelegate(contractAddress, NON_OWNER, 0x1);
-        blockNumberForRemoval = (await provider.getBlockNumber()) + 5;
-        await removeDelegate(contractAddress, NON_OWNER, blockNumberForRemoval);
+    describe("when delegate is added removed and re-added", () => {
+      const provider = ({
+        getBlockNumber: jest.fn().mockResolvedValue(400),
+      } as Partial<ethers.providers.Provider>) as ethers.providers.Provider;
+
+      const delegateAddress = "0x70997970C51812dc3A010C7d01b50e0d17dc79C8";
+
+      const delegateLogData: DelegateLogData[] = [
+        {
+          name: "DSNPAddDelegate",
+          identityAddress: "0x71C95911E9a5D330f4D621842EC243EE1343292e",
+          delegate: "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+          blockNumber: 330,
+        },
+        {
+          name: "DSNPRemoveDelegate",
+          identityAddress: "0x71C95911E9a5D330f4D621842EC243EE1343292e",
+          delegate: "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
+          blockNumber: 331,
+          endBlock: 0,
```

Comment:
This would be more interesting if `endBlock` were 333. I believe it would do the same thing and I believe (with less confidence) that the same behavior is the correct behavior. Consider changing this to highlight that fact.

Reply:
done. I had made it zero to make it obvious. I updated it.